From cf8f5a4e4098a6aae9b986dc2da2006eadd4fef1 Mon Sep 17 00:00:00 2001
From: Niels de Vos <ndevos@redhat.com>
Date: Thu, 18 Jun 2015 12:16:16 +0200
Subject: [PATCH 13/74] rpm: glusterfs-devel for client-builds should not
 depend on -server

glusterfs-devel for client-side packages should *not* include the
libgfdb.so symlink and libgfdb.pc file or any of the libchangelog
ones.

Label: DOWNSTREAM ONLY

Change-Id: Ifb4a9cf48841e5af5dd0a98b6de51e2ee469fc56
Signed-off-by: Niels de Vos <ndevos@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/51019
Reviewed-by: Balamurugan Arumugam <barumuga@redhat.com>
Tested-by: Balamurugan Arumugam <barumuga@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/60139
Tested-by: Milind Changire <mchangir@redhat.com>
---
 glusterfs.spec.in | 24 +++++++++++++++++++-----
 1 file changed, 19 insertions(+), 5 deletions(-)

diff --git a/glusterfs.spec.in b/glusterfs.spec.in
index 68eba56..b2fb4d5 100644
--- a/glusterfs.spec.in
+++ b/glusterfs.spec.in
@@ -1196,9 +1196,10 @@ exit 0
 %{_tmpfilesdir}/gluster.conf
 %endif
 %if ( ! 0%{?_build_server} )
-%{_libdir}/pkgconfig/libgfchangelog.pc
-%{_libdir}/pkgconfig/libgfdb.pc
-%{_sbindir}/gluster-setgfid2path
+%exclude %{_libdir}/pkgconfig/libgfchangelog.pc
+%exclude %{_libdir}/pkgconfig/libgfdb.pc
+%exclude %{_sbindir}/gluster-setgfid2path
+%exclude %{_mandir}/man8/gluster-setgfid2path.8*
 %endif
 
 %files api
@@ -1226,6 +1227,12 @@ exit 0
      %{_includedir}/glusterfs/*
 %exclude %{_includedir}/glusterfs/api
 %exclude %{_libdir}/libgfapi.so
+%if ( ! 0%{?_build_server} )
+%exclude %{_libdir}/libgfchangelog.so
+%endif
+%if ( 0%{!?_without_tiering:1} && ! 0%{?_build_server})
+%exclude %{_libdir}/libgfdb.so
+%endif
 %{_libdir}/*.so
 # Glupy Translator examples
 %dir %{_libdir}/glusterfs/%{version}%{?prereltag}/xlator
@@ -1235,10 +1242,14 @@ exit 0
      %{_libdir}/glusterfs/%{version}%{?prereltag}/xlator/features/glupy/helloworld.*
      %{_libdir}/glusterfs/%{version}%{?prereltag}/xlator/features/glupy/negative.*
 %if ( 0%{?_build_server} )
+%{_libdir}/pkgconfig/libgfchangelog.pc
+%else
 %exclude %{_libdir}/pkgconfig/libgfchangelog.pc
+%endif
+%if ( 0%{!?_without_tiering:1} && 0%{?_build_server})
+%{_libdir}/pkgconfig/libgfdb.pc
+%else
 %exclude %{_libdir}/pkgconfig/libgfdb.pc
-%exclude %{_sbindir}/gluster-setgfid2path
-%exclude %{_mandir}/man8/gluster-setgfid2path.8*
 %endif
 
 %files client-xlators
@@ -2161,6 +2172,9 @@ end
 * Tue Aug 18 2015 Niels de Vos <ndevos@redhat.com>
 - Include missing directories for glusterfind hooks scripts (#1225465)
 
+* Thu Jun 18 2015 Niels de Vos <ndevos@redhat.com>
+- glusterfs-devel for client-builds should not depend on -server (#1227029)
+
 * Mon Jun 15 2015 Niels de Vos <ndevos@redhat.com>
 - Replace hook script S31ganesha-set.sh by S31ganesha-start.sh (#1231738)
 
-- 
1.8.3.1

